woocommerce/woocommerce-gutenberg-products-block
# Product Button: Remove the hardcoded width and padding

## What

This PR removes the hardcoded button width and padding (to inherit the padding from the theme styling).

Fixes #11436

## Why

Currently, the width of the buttons is hardcoded to `150px`. My assumption is that this was introduced to prevent the following effect of "jumping width" because of the spinner:
![279669403-8c05f08c-6101-44f8-b715-7cbe171b7910](https://github.com/woocommerce/woocommerce-blocks/assets/905781/f7dfcf11-1a7e-410d-b9ec-c0a99a7dec71)

The hardcoded width is less than ideal as it leads to issues like #11436.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Add the `4-Column Product Row` pattern to a post or a page.
2. Go to the front end and try re-sizing the viewport.
3. Make sure the buttons are not overflowing.
4. Try adding products to the cart. Make sure there's no odd visual behavior.
5. Go to the `Shop` page and make sure the buttons look and behave correctly.
6. Try testing with regular, wide and full width.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|<img width="754" alt="4-Column_Product_Row_–_newproductgallery" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/7ee201f1-e2de-4bd7-861c-7f63a7131daa">|<img width="702" alt="4-Column_Product_Row_–_newproductgallery" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/3d80f974-0d61-4c02-a600-99e0d17abd43">|

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Button: Improve the width and padding.
